src/main/scala/it/pagopa/interop/purposeprocess/api/impl/PurposeApiServiceImpl.scala

```diff
@@ -108,9 +108,27 @@ final case class PurposeApiServiceImpl(
         seed.consumerId,
         OPERATIVE_AGREEMENT_STATES
       )
-      _       <- agreements.headOption.toFuture(AgreementNotFound(seed.eserviceId.toString, seed.consumerId.toString))
-      purpose <- purposeManagementService.createPurpose(clientSeed)
+      agreement <- agreements.headOption.toFuture(AgreementNotFound(seed.eserviceId.toString, seed.consumerId.toString))
+      maybePurpose <- ReadModelQueries
+        .listPurposes(
+          seed.consumerId,
+          seed.title.some,
+          List(agreement.eserviceId.toString),
+          List(agreement.consumerId.toString),
+          List(agreement.producerId.toString),
+          List.empty,
+          false,
+          0,
+          1,
```

Comment:
Just for readability
```suggestion
          states = List.empty,
          excludeDraft = false,
          offset = 0,
          limit = 1,
```